src/handlers/register/culdap_register_handler.py

```diff
@@ -0,0 +1,176 @@
+
+    def get(self):
+        return self.render("register/culdapregister.html", errors=[], next=self.get_argument("next","/"))
+
+
+    def post(self):
+        confirming = self.get_argument('confirming',False,strip = True)
+        if confirming:
+            return self.confirmCULdapRegistration()
+        else:
+            return self.CULdapRegister()
+
+
+    def confirmCULdapRegistration(self):
+        username = self.get_argument('username',strip = True)
+        errors = self.getVerificationErrors()
+        if len(errors) != 0:
+            return self.verifyCULdapRegistrationPage(username, errors)
+        cookie_username = self.get_secure_cookie(self.COOKIE, max_age_days=self.FIVE_MINUTES)
+        if cookie_username is None:
+            return self.failWithErrors('register/culdapregister.html', ['Time to complete registration expired (5 minutes)'])
+        decoded_cookie_username = cookie_username.decode("utf-8")
+        if username != decoded_cookie_username:
+            logging.error('username %s did not match cookie username %s' % (username, decoded_cookie_username))
+            return self.failWithErrors('register/culdapregister.html', ['Registration failed: cookie data is invalid'])
+        data = self.collectUserData()
+        data['username']        = username
+        data['registration']    = User().REGISTRATION_CULDAP
+        data['accepted_tos']    = True
+        return self.registerUser(data)
+
+    def collectUserData(self):
+        data = {}
+        data['email']           = self.get_argument('email',None,strip = True)
+        data['dob']             = self.get_argument('dob',None,strip = True)
+        data['gender']          = self.get_argument('gender',None,strip = True)
+        data['ethnicity']       = self.get_argument('ethnicity',None,strip = True)
+        data['native_language'] = self.get_argument('native_language',None,strip = True)
+        data['status']          = self.get_argument('status',None,strip = True)
+        data['major1']          = self.get_argument('major1',None,strip = True)
+        data['major2']          = self.get_argument('major2',None,strip = True)
+        data['major3']          = self.get_argument('major3',None,strip = True)
+        data['major4']          = self.get_argument('major4',None,strip = True)
+        data['minor1']          = self.get_argument('minor1',None,strip = True)
+        data['minor2']          = self.get_argument('minor2',None,strip = True)
+        return data
+
+
+    def registerUser(self,data):
+        data['date_registered'] = time.strftime('%a %b %d %H:%M:%S %Z %Y')
+        verified = User().verify(data)
+        if len(verified) != 0:
+            logging.error('User: verification errors!')
+            logging.error(verified)
+            return self.verifyCULdapRegistrationPage(username, verified)
+        user_id = User().create_item(data)
+        # user_id = yield r.db(BaseModel.DB).table("User").insert(data).run(BaseModel.conn)
```

Comment:
What line? 82? Keep that, I'm still working on registration. All thats left to be done is give the user a high five for registering and kick them back to index / wherever they were browsing.